src/ripple/protocol/impl/STObject.cpp

```diff
@@ -112,43 +122,39 @@ bool STObject::setType (const SOTemplate& type)
                 { return b.get().getFName() == e->e_field; });
         if (iter != v_.end())
         {
-            if ((e->flags == SOE_DEFAULT) && iter->get().isDefault())
+            if ((e->flags == SOE_DEFAULT) &&
+                iter->get().isDefault() &&
+                result.empty())
```

Comment:
So, this causes a slight change in this function's logging behavior: previously we'd log _every_ bad field whereas now we only log the first error.

That's actually fine by me. In fact, my personal preference would be to just call it a day on the very first indication that the object is invalid by doing a `throw` and indicating the problematic field.

Reply:
Right.  I recognized the change when I was making it.  I felt like logging (and formatting the error string for) one error was sufficient.  And reporting only the first error lowers the cost of handling cases where there is a mismatch between the template and the `STObject`.

I considered stopping processing altogether on the first error, but I chose not to do that because it would be a significant change to the behavior of this `STObject` constructor: https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/impl/STObject.cpp#L66